libgeopmd/src/StatsCollector.cpp

```diff
@@ -61,6 +65,28 @@ namespace geopm
     void StatsCollector::update(void)
     {
         // Caller is expected to have called platform_io().read_batch();
+        ++m_sample_count;
+        double time_last = m_time_sample;
+        m_time_sample = m_pio.sample(m_time_pio_idx);
+        if (m_time_begin_str == "") {
+            m_time_begin = m_time_sample;
+            double time_curr = m_pio.read_signal("TIME", GEOPM_DOMAIN_BOARD, 0);
+            geopm_time_s time_curr_real = geopm::time_curr_real();
+            geopm_time_s time_begin_real;
+            geopm_time_add(&time_curr_real, m_time_sample - time_curr, &time_begin_real);
+            char time_begin_cstr[NAME_MAX];
+            int err = geopm_time_real_to_iso_string(&time_begin_real, NAME_MAX, time_begin_cstr);
+            if (err != 0) {
+                throw Exception("StatsCollector::update(): geopm_time_real_to_iso_string() call failed",
+                                err, __FILE__, __LINE__);
+            }
+            m_time_begin_str = time_begin_cstr;
+        }
+        else {
+            double time_delta = m_time_sample - time_last;
+            m_time_delta_m_1 += time_delta;
+            m_time_delta_m_2 += time_delta * time_delta;
+        }
```

Comment:
I'm not sure, to me seems more confusing.  Maybe i need to document that m_sample_count is the number of times sample() has been called? 

Because we are doing time differences we loose a degree of freedom compared to the total number of time samples that have been made (m_sample_count).  This extra degree of freedom lost is why the denominators is smaller.  If m_sample_count were used in the future to take the average of any other sampled value it would be off by one.  In this use case (of course), the code change is equivalent.